### ISSUE TITLE
BigQueryClient ignores manually set scope from credentials

### ISSUE THREAD

**Issue by tazy7** [googleapis/google-cloud-dotnet#5177]
I am using the latest version of the package Google.Cloud.BigQuery.V2 2.0.0, when trying to query federated tables, Google Sheet backed data I was getting an OAuth exception expecting the Google Drive scope to be supplied.

I have both tried inside of a ServiceAccountCredential set the scopes, and on the GoogleCredential wrapper call CreateScoped the supplied scopes are simply ignored. This seems to only happen when using the BigQueryClient, the client is returning 4 scopes back rather than the 2 I supply, ignoring the drive token. However if I manually setup the BigqueryService with an initialiser the Scopes supplied are respected and used.

Below are some snippets of the code I am using.

`private` static readonly string[] Scopes = { "https://www.googleapis.com/auth/drive", "https://www.googleapis.com/auth/bigquery" };

var client = BigQueryClient.Create(queryDetails.BigQueryProject, GoogleCredential.FromServiceAccountCredential(GetCredential()))

            return new ServiceAccountCredential(
                new ServiceAccountCredential.Initializer(credentialParameters.ClientEmail)
                {
                    Scopes = Scopes
                }.FromPrivateKey(credentialParameters.PrivateKey));
`

**Comment by amanda-tarafa** [comment]
The behaviour you are seeing is by design. Both the [`BigQueryClient.Create`](https://googleapis.dev/dotnet/Google.Cloud.BigQuery.V2/2.0.0/api/Google.Cloud.BigQuery.V2.BigQueryClient.html#Google_Cloud_BigQuery_V2_BigQueryClient_Create_System_String_Google_Apis_Auth_OAuth2_GoogleCredential_) and [`BigQueryClient.CreateAsync`](https://googleapis.dev/dotnet/Google.Cloud.BigQuery.V2/2.0.0/api/Google.Cloud.BigQuery.V2.BigQueryClient.html#Google_Cloud_BigQuery_V2_BigQueryClient_CreateAsync_System_String_Google_Apis_Auth_OAuth2_GoogleCredential_) methods will scope the credentials with some default BigQuery scopes. It's documented on the Remarks sections of the links above. I understand this might be a little unexpected but in any case we wouldn't want to change that behaviour so as not to break anyone relying on it.

You need to use [`BigQueryClientBuilder`](https://googleapis.dev/dotnet/Google.Cloud.BigQuery.V2/2.0.0/api/Google.Cloud.BigQuery.V2.BigQueryClientBuilder.html) and set the `Credential` property before calling `Build()` which will return a `BigQueryClient`. 
For creating the scoped credential, you can use code similar to yours:

```
GoogleCredential credential = GoogleCredential.FromServiceAccountCredential(
    new ServiceAccountCredential(
        new ServiceAccountCredential.Initializer(email).FromPrivateKey(privateKey)))
    .CreateScoped(scopes);
```

or if you have the service account key in a json file (there's also the async version FromFileAsync):

`GoogleCredential credential = GoogleCredential.FromFile(jsonFilePath).CreateScoped(scopes);`

Once you have the scoped credential:

```
BigQueryClientBuilder builder = new BigQueryClientBuilder { Credential = credential };
BigQueryClient client = builder.Build();
```

**Comment by tazy7** [comment]
Thank you for replying back to me, is it possible to make this clearer in the documentation that the use of CreatedScopes will be ignored unless using the clientbuilder? 

The remark of "The credentials are scoped as necessary." is not overly clear in this case, to me it means it gives the basic scopes, not it can't take an override behaviour.

I have looked in the documentation for nearly a full day to see if I had missed this. I only discovered this was an issue with the c# implementation after testing against a working NodeJs code base with the same federated table.
